Add subscription model

IslamGoher opened the pull request:
- Create schema of subscription
- Add messages of validation errors
- Add relations between subscription model with all of guardian, child, mentor and course models

Commit:
Add subscription model

- Create schema of subscription
- Add messages of validation errors
- Add relations between subscription model with all of guardian, child, mentor and course models

mohmedeprahem commented on models/subscriptions.js:
```diff
@@ -0,0 +1,84 @@
+      },
+      date: {
+        type: Date,
+        default: Date.now()
```
delete default

mohmedeprahem commented on models/subscriptions.js:
```diff
@@ -0,0 +1,84 @@
+        type: mongoose.Schema.Types.ObjectId,
+        default: mongoose.Types.ObjectId()
+      },
+      date: {
```
add required = true

mohmedeprahem commented on models/subscriptions.js:
```diff
@@ -0,0 +1,84 @@
+        type: mongoose.Schema.Types.ObjectId,
+        required: [true, `please add appoinement id.`]
+      },
+      date: Date,
```
add required = true in date property

Commit:
Edit subscription model

- Made date of appointment required
- Remove default value to date of appointment
- Made date of notes required

fix #2

mohmedeprahem reviewed (changes requested):
please fix bugs
